### ISSUE TITLE
JS & CSS Combine

### ISSUE THREAD

**Issue by WPLibrary** [W3EDGE/w3-total-cache#170]
**With current options in auto minification:**
- There is no "minify only" option,
- You have to combine JS in order to defer it,
- To enable the HTTP/2 Push option, you must activate "combine" first.

Manual minify feature is absolutely brilliant and W3TC is the only plugin which offers that option. We can use the "minify only" option if the Minify Mode is manual. However, As theme&plugin assets are frequently changing, people usually tend to use the "Auto minify" mode I think. Like I said we don't have the above features if the minify mode is Auto.

I think, more customization in the minify settings will make the plugin better.

After HTTP/2 improvements, both JS or CSS combine features do NOT have a significant role on page load. Even HTTP/3 is also coming...

Regards.

**Comment by WPLibrary** [comment]
Also, a review about JS combine from wordpress.org (AJ @ WPFaster):

> If your website is served over HTTP/2 and has ‘lots’ of JavaScript, concatenating this JavaScript into a single or even a few files will decrease real world performance.
> 
> Multiple minified and deferred JavaScript files are parsed far more efficiently over HTTP/2 since they are multiplexed/served in parallel, whereas if one has them all — or nearly all — of them combined, the end user is going to have to wait the additional time it takes for that one big JavaScript file to finish. The real world affect being the delayed loading of anything above the fold or functionality dependent upon JavaScript, sliders for example. That a combined JavaScript file can be deferred does not ipso facto translate into improved perceived speed, even if some or another best practices recommendations tool gives one higher grades for having combined their website’s JavaScript.
> 
> In short, while tools like GTmetrix can be helpful if interpreted correctly, the bottom line is that blindly following the suggestions tools like this make will often result in anti-patterns which translate into a slower site for one’s real human users, which then translates into degraded UX, which then of course translates into the Universe collapsing in on itself and the end of all life in the Cosmos… please don’t end all life in the Cosmos.

**Comment by maxicus** [comment]
Thanks for sharing. 
At first I want to say that this feature is fine in general.

Still I'll leave my 2 cents while I pass that over.

While it's true theory says it should be avoided, I still see that combining helps A LOT in practical world even for http/2. Seems overhead for browser and max multiplexed level, hopefully that will change in future.

So... be very careful sharing info above. Good test cases will be good to show when it happens and when not. Note that mine sentence above is equally naked and far from being ideal :)

**Comment by lwj5** [comment]
@maxicus I've added the PR, hope you could help review it. 

Also, I've posted a question about key migration and deprecation in the PR. Let me know how I can sort that out.
